utils/slither_format/format_external_function.py

```diff
@@ -0,0 +1,55 @@
+import re, logging
+from slither.utils.colors import red, yellow, set_colorization_enabled
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger('Slither.Format')
+set_colorization_enabled(True)
+
+class FormatExternalFunction:
+
+    @staticmethod
+    def format (slither, patches, elements):
+        for element in elements:
+            Found = False
+            for contract in slither.contracts:
+                if not Found and contract.name == element['type_specific_fields']['parent']['name']:
+                    for function in contract.functions:
+                        if function.name == element['name']:
+                            # If function parameters are written to in function body then we cannot convert this function
+                            # to external because external function parameters are allocated in calldata region which is
+                            # non-modifiable. See https://solidity.readthedocs.io/en/develop/types.html#data-location
+                            if not FormatExternalFunction.function_parameters_written(function):
+                                FormatExternalFunction.create_patch(slither, patches, element['source_mapping']['filename_absolute'], "public", "external", int(function.parameters_src.source_mapping['start']), int(function.returns_src.source_mapping['start']))
+                                Found = True
+                                break
+
+    @staticmethod
+    def create_patch(slither, patches, in_file, match_text, replace_text, modify_loc_start, modify_loc_end):
+        in_file_str = slither.source_code[in_file]
+        old_str_of_interest = in_file_str[modify_loc_start:modify_loc_end]
+        m = re.search(" public", old_str_of_interest)
```

Comment:
Consider using `" public "`, to prevent collision with the name of the function

Reply:
Made the regex stricter. The prefix collision could be with the modifier if present.